internal/backend/runtime/omni/controllers/omni/machine_set_node.go

```diff
@@ -175,6 +188,28 @@ func (ctrl *MachineSetNodeController) Run(ctx context.Context, r controller.Runt
 	}
 }
 
+// handleMachinePress.
+func (ctrl *MachineSetNodeController) handleRequiredAdditionalMachines(ctx context.Context, r controller.Runtime, machineSet *omni.MachineSet, numRequired int) error {
+	if numRequired > 0 {
+		return safe.WriterModify[*omni.MachineSetRequiredMachines](ctx, r, omni.NewMachineSetRequiredMachines(resources.DefaultNamespace, machineSet.Metadata().ID()),
+			func(res *omni.MachineSetRequiredMachines) error {
+				helpers.CopyAllLabels(machineSet, res)
+
+				res.Metadata().Labels().Set(omni.LabelMachineClassName, machineSet.TypedSpec().Value.MachineClass.Name)
+
+				res.TypedSpec().Value.RequiredAdditionalMachines = uint32(numRequired)
+
+				return nil
+			})
+	}
+
+	if err := r.Destroy(ctx, omni.NewMachineSetRequiredMachines(resources.DefaultNamespace, machineSet.Metadata().ID()).Metadata()); err != nil && !state.IsNotFoundError(err) {
```

Comment:
As we destroy machine set required machines if the number == 0 we need to add the finalizer on resource in the MachineClassStatus controller.
Then we'll need to run the Teardown + Destroy flow.

I think maybe we can avoid all that if we just set the number of required machines to 0 in the resource and do not remove it.

Reply:
👍  I applied the change you suggested in this controller.

But ofc, still, `MachineSetController` (the `cleanup` controller) removes them when the corresponding `MachineSet` is removed.